stdlib/bigarray.ml

```diff
@@ -115,7 +115,6 @@ module Genarray = struct
   let init (type t) kind (layout : t layout) dims f =
     let arr = create kind layout dims in
     match Array.length dims, layout with
-    | 0, _ -> arr
     | dlen, C_layout -> cloop arr (Array.make dlen 0) f 0 dims; arr
     | dlen, Fortran_layout -> floop arr (Array.make dlen 1) f (pred dlen) dims;
                               arr
```

Comment:
Now that there's no branch on `dlen`, it could be pulled out of the `match`:

```ocaml
  let init (type t) kind (layout : t layout) dims f =
    let arr = create kind layout dims in
    let dlen = Array.length dims in
    match layout with
    | C_layout       -> cloop arr (Array.make dlen 0) f          0  dims; arr
    | Fortran_layout -> floop arr (Array.make dlen 1) f (pred dlen) dims; arr
```

Reply:
Right; fixed. Thanks.